### PR TITLE
remove unused var from deleted stuff

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: include/exiv2/basicio.hpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ class EXIV2API BasicIo {
     @return Number of bytes written to IO source successfully;<BR>
         0 if failure;
    */
-  virtual size_t write(const byte *data, size_t wcount) = 0;
```
Comment (piponazo) [inline include/exiv2/basicio.hpp:82]: O_O ... is not this controlled by clang-format? I am pretty sure the is something in clang-format to control the indentation of `&` and `*` in function parameters.

Was this something that you changed manually?

File: include/exiv2/datasets.hpp [hunk 2]
```diff
@@ -132,12 +132,10 @@ class EXIV2API IptcDataSets {
   static constexpr uint16_t Preview = 202;
   //@}
 
-  //! Prevent construction: not implemented.
-  IptcDataSets() = delete;
```
Comment (piponazo) [inline include/exiv2/datasets.hpp:136]: What's the reasoning to allow the creation of this class now? This change looks out of the scope of the commit.

File: include/exiv2/jpgimage.hpp [hunk 3]
```diff
@@ -104,12 +104,10 @@ class EXIV2API JpegBase : public Image {
 
   //! @name NOT implemented
   //@{
-  //! Default constructor.
-  JpegBase() = delete;
```
Comment (piponazo) [inline include/exiv2/jpgimage.hpp:108]: Same comment in all the default constructors here. We should not allow the creation of a instance via the default constructor.

File: include/exiv2/datasets.hpp (hunk 2 above)
Comment (neheb) [reply include/exiv2/datasets.hpp:136]: A default constructor does not get generated here. I removed it as it's a bit misleading.

File: include/exiv2/basicio.hpp (hunk 1 above)
Comment (neheb) [reply include/exiv2/basicio.hpp:82]: clang-format did this, yes. In the current setup, clang-format decides the order based on the most used order. Probably should disable DerivePointerAlignment.

Comment (piponazo) [reply include/exiv2/basicio.hpp:82]: I agree, probably we should use something like:

```
# Force pointers to the type for C++.
DerivePointerAlignment: false
PointerAlignment: Left
```
To enforce the same style over the whole project. @hassec what's your opinion on this point? 

For me it is fine to accept this PR as it is, and then create another PR to address those clang-format changes.

File: include/exiv2/datasets.hpp (hunk 2 above)
Comment (piponazo) [reply include/exiv2/datasets.hpp:136]: Oh god, you're totally right. I am too far from coding lately and I forget these details about C++ 😩

**Review by piponazo — APPROVED** [review]
LGTM, thanks!

File: include/exiv2/basicio.hpp (hunk 1 above)
Comment (hassec) [reply include/exiv2/basicio.hpp:82]: Fully agree with you  @piponazo :+1:       
And yes, I also think we can move this forward independently and fix `clang-format` in a different PR

Comment (neheb) [reply include/exiv2/basicio.hpp:82]: Pointer alignment should be left, yes.

File: include/exiv2/datasets.hpp (hunk 2 above)
Comment (hassec) [reply include/exiv2/datasets.hpp:136]: I think leaving the explicit `= delete` doesn't hurt though and helps a lot when reading the code as one then doesn't need to worry about knowing the details of when special member functions are implicitly declared...

Comment (piponazo) [reply include/exiv2/datasets.hpp:136]: I also think that sometimes being explicit about certain things is better for code readability, but it is difficult to define where to draw the line. On the other hand, less code is almost always the way to go for me. In this case, I think it is fine to remove those lines, if somebody try to create an instance of the class with the default constructor the compiler will complain anyways.

**Pull request merged**